File: src/main/java/com/epri/dss/general/impl/WireDataObjImpl.java
package com.epri.dss.general.impl;

import java.io.PrintStream;

import com.epri.dss.common.DSSClass;
import com.epri.dss.general.WireData;
import com.epri.dss.general.WireDataObj;
import com.epri.dss.shared.impl.LineUnits;

public class WireDataObjImpl extends ConductorDataObjImpl implements WireDataObj {

	public WireDataObjImpl(DSSClass parClass, String wireDataName) {
		super(parClass, wireDataName);
	}

	@Override
	public void dumpProperties(PrintStream f, boolean complete) {
		super.dumpProperties(f, complete);

		for (int i = 0; i < parentClass.getNumProperties(); i++) {
			f.print("~ " + parentClass.getPropertyName()[i] + "=");
			switch (i) {
			case 0:
				f.println(String.format("%.6g", getRDC()));
				break;
			case 1:
				f.println(String.format("%.6g", getR60()));
				break;
			case 2:
				f.println(String.format("%s", LineUnits.lineUnitsStr(getResistanceUnits())));
				break;
			case 3:
				f.println(String.format("%.6g", getGMR60()));
				break;
			case 4:
				f.println(String.format("%s", LineUnits.lineUnitsStr(getGMRUnits())));
				break;
			case 5:
				f.println(String.format("%.6g", getRadius()));
				break;
			case 6:
				f.println(String.format("%s", LineUnits.lineUnitsStr(getRadiusUnits())));
				break;
			case 7:
				f.println(String.format("%.6g", getNormAmps()));
				break;
			case 8:
				f.println(String.format("%.6g", getEmergAmps()));
				break;
			case 9:
				f.println(String.format("%.6g", getRadius() * 2.0));
				break;
			}
		}
	}

	@Override
	public void initPropertyValues(int arrayOffset) {
		propertyValue[1] = "-1";
		propertyValue[2] =  "-1";
		propertyValue[3] =  "none";
		propertyValue[4] =  "-1";
		propertyValue[5] =  "none";
		propertyValue[6] =  "-1";
		propertyValue[7] =  "none";
		propertyValue[8] =  "-1";
		propertyValue[9] =  "-1";
		propertyValue[10] =  "-1";

		super.initPropertyValues(WireData.NumPropsThisClass);
	}
}

File: pydss/general/WireDataObjImpl.py
from dss.general.WireData import (WireData,)
from dss.shared.impl.LineUnits import (LineUnits,)
from dss.general.WireDataObj import (WireDataObj,)
from dss.general.impl.ConductorDataObjImpl import (ConductorDataObjImpl,)


class WireDataObjImpl(ConductorDataObjImpl, WireDataObj):

    def __init__(self, parClass, wireDataName):
        super(WireDataObjImpl, self)(parClass, wireDataName)

    def dumpProperties(self, f, complete):
        super(WireDataObjImpl, self).dumpProperties(f, complete)
        _0 = True
        i = 0
        while True:
            if _0 is True:
                _0 = False
            else:
                i += 1
            if not (i < self.parentClass.getNumProperties()):
                break
            f.print_('~ ' + self.parentClass.getPropertyName()[i] + '=')
            _1 = i
            _2 = False
            while True:
                if _1 == 0:
                    _2 = True
                    print String.format.format('%.6g', self.getRDC())
                    break
                if (_2 is True) or (_1 == 1):
                    _2 = True
                    print String.format.format('%.6g', self.getR60())
                    break
                if (_2 is True) or (_1 == 2):
                    _2 = True
                    print String.format.format('%s', LineUnits.lineUnitsStr(self.getResistanceUnits()))
                    break
                if (_2 is True) or (_1 == 3):
                    _2 = True
                    print String.format.format('%.6g', self.getGMR60())
                    break
                if (_2 is True) or (_1 == 4):
                    _2 = True
                    print String.format.format('%s', LineUnits.lineUnitsStr(self.getGMRUnits()))
                    break
                if (_2 is True) or (_1 == 5):
                    _2 = True
                    print String.format.format('%.6g', self.getRadius())
                    break
                if (_2 is True) or (_1 == 6):
                    _2 = True
                    print String.format.format('%s', LineUnits.lineUnitsStr(self.getRadiusUnits()))
                    break
                if (_2 is True) or (_1 == 7):
                    _2 = True
                    print String.format.format('%.6g', self.getNormAmps())
                    break
                if (_2 is True) or (_1 == 8):
                    _2 = True
                    print String.format.format('%.6g', self.getEmergAmps())
                    break
                if (_2 is True) or (_1 == 9):
                    _2 = True
                    print String.format.format('%.6g', self.getRadius() * 2.0)
                    break
                break

    def initPropertyValues(self, arrayOffset):
        self.propertyValue[1] = '-1'
        self.propertyValue[2] = '-1'
        self.propertyValue[3] = 'none'
        self.propertyValue[4] = '-1'
        self.propertyValue[5] = 'none'
        self.propertyValue[6] = '-1'
        self.propertyValue[7] = 'none'
        self.propertyValue[8] = '-1'
        self.propertyValue[9] = '-1'
        self.propertyValue[10] = '-1'
        super(WireDataObjImpl, self).initPropertyValues(WireData.NumPropsThisClass)
